### PR TITLE
Update docs: Note RMSprop usage instead of Adam for memory savings in…

### PR DESCRIPTION
**Pull request by nasyxx**
… schedule_free

This commit updates the documentation to reflect the use of RMSprop optimizer instead of Adam, addressing the memory conservation issue raised in #1010.

### CODE REVIEW

**Review by vroulet — COMMENTED** [review]
Thanks for doing that!

File: optax/contrib/_schedule_free.py [hunk 1]
```diff
@@ -312,7 +312,8 @@ def schedule_free_adamw(
   Shortcut example for using schedule_free with AdamW, which is a common use
   case. Note that this is just an example, and other usecases are possible, e.g.
   using a weight decay mask, nesterov, etc. Note also that the EMA parameter of
-  the schedule free method (b1) must be strictly positive.
+  the schedule free method (b1) must be strictly positive. Due to issue #1010,
```
[the remark this answers is not in the document]
Comment (vroulet) [reply optax/contrib/_schedule_free.py:315]: Then add a screenshot of the resulting doc just as Fabian did in #1085

**Comment by nasyxx** [comment]
Screenshot 2024-10-02 at 2 19 41 PM

The screenshot.

**Comment by nasyxx** [comment]
Screenshot 2024-10-02 at 5 31 09 PM
The new screenshot.

Do I need to make all commits into one?

**Comment by vroulet** [comment]
If you can squash your commits that'll be great.
You may want to first fix the linting issue (some lines are too long). 
Thank you again.

**Review by vroulet — APPROVED** [review]

**Pull request merged**